src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataScriptTempFileStream.cs

```diff
@@ -0,0 +1,123 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.SqlTools.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.Metadata
+{
+    /// <summary>
+    /// This class is responsible for reading, writing, and checking the validity of script files.
+    /// </summary>
+    public static class MetadataScriptTempFileStream
+    {
+        private const short ScriptFileExpirationInDays = 30;
+
+        /// <summary>
+        /// This method writes the passed in scripts to a temporary file.
+        /// </summary>
+        /// <param name="serverName">The name of the server which will go on to become the name of the file.</param>
+        /// <param name="scripts">The generated scripts that will be written to the temporary file.</param>
+        public static void Write(string serverName, IEnumerable<string> scripts)
+        {
+            var tempFileName = $"{serverName}.tmp";
```

Comment:
Server names can contain invalid filename characters (like `/`), so you should probably be sanitizing this before reading/writing. Probably easiest to just convert to base64 or something.